scripts/mbedtls_dev/crypto_knowledge.py

```diff
@@ -0,0 +1,51 @@
+"""Knowledge about cryptographic mechanisms implemented in Mbed TLS.
+
+This module is entirely based on the PSA API.
+"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from typing import List, Optional
+
+class KeyType:
+    """Knowledge about a PSA key type."""
+
+    def __init__(self, name: str, params: Optional[List[str]] = None):
+        """Analyze a key type.
+
+        The key type must be specified in PSA syntax. In its simplest form,
+        this is a string 'PSA_KEY_TYPE_xxx' which is the name of a PSA key
+        type macro. For key types that take arguments, the arguments can
+        be passed either through the optional argument `params` or by
+        passing an expression of the form 'PSA_KEY_TYPE_xxx(param1, param2)'
+        as the a string.
+        """
+        self.name = name.strip()
```

Comment:
I don't know what checks would be useful here. `startswith('PSA_KEY_TYPE_')`? Or `'PSA_KEY_TYPE_' in` so that we can add `MBEDTLS_PSA_KEY_TYPE_xxx`? I don't see what `PUBLIC_KEY` and `KEY_PAIR` have to do here: this function deals with all key types.

Reply:
I was just thinking at basic sanity checks of the name. In mentioning PUBLIC_KEY and KEY_PAIR I forgot about the symmetric keys.

Reply:
What kinds of usage mistakes do you expect?

Reply:
That `name` is not a string of type `PSA_KEY_TYPE_xxx`.

Reply:
I'm not convinced that it's useful, but ok, I added a check on the name.